autopilot/contractor.go

```diff
@@ -452,20 +455,47 @@ func (c *contractor) performWalletMaintenance(ctx context.Context) error {
 	c.logger.Info("performing wallet maintenance")
 	b := c.ap.bus
 	l := c.logger
+	state := c.ap.State()
 
 	// no contracts - nothing to do
-	cfg := c.ap.State().cfg
-	if cfg.Contracts.Amount == 0 {
+	if state.cfg.Contracts.Amount == 0 {
 		l.Warn("wallet maintenance skipped, no contracts wanted")
 		return nil
 	}
 
 	// no allowance - nothing to do
-	if cfg.Contracts.Allowance.IsZero() {
+	if state.cfg.Contracts.Allowance.IsZero() {
 		l.Warn("wallet maintenance skipped, no allowance set")
 		return nil
 	}
 
+	// fetch wallet balance
+	wallet, err := b.Wallet(ctx)
+	if err != nil {
+		l.Errorf("wallet maintenance skipped, fetching wallet balance failed with err: %v", err)
+		return err
+	}
+	balance := wallet.Spendable
+
+	// register an alert if balance is low
+	min, max := initialContractFundingMinMax(state.cfg)
+	if balance.Cmp(max) <= 0 {
```

Comment:
Isn't this very optimistic? If you have a full set with data that set might fail to refresh or renew because by then more money than `max` is required to refresh/renew.

Reply:
hm ok so I checked on arequipa and it is indeed a bit optimistic, I updated it